js/plugin.iiif-iabookreader_strawberry.js

```diff
@@ -215,6 +249,44 @@ BookReader.prototype.parseSequence = function (sequenceId) {
     return imagesList;
   }
 
+  function getImagesListApi3(items) {
+    var imagesList = [];
+
+    jQuery.each(items, function (index, item) {
+      if (item['type'] === 'Canvas') {
+        let imageObj = {
+          canvasHeight: item.height || 0,
+          canvasWidth: item.width || 0,
+        };
+        let annotationpages = item.items;
+        jQuery.each(annotationpages, function (index, annotationpage) {
+          if ((annotationpage['type'] === 'AnnotationPage') && (annotationpage['items'][0]['type'] === 'Annotation') && (annotationpage['items'][0]['body'])) {
+            let annotation = annotationpage['items'][0];
+            imageObj.serviceUrl = null;
+            if (annotation.body.hasOwnProperty('service') && isValidHttpUrl(annotation.body.service['id'])) {
+              imageObj.serviceUrl = annotation.body.service['id'].replace(/\/$/, '');
```

Comment:
pretty sure this is where it is failing...  service is an array.. (might have multiple entries) and each one has an id
`isValidHttpUrl` might be giving you a false positive ...
``` JSON
"service": [
                            {
                                "id": "http:\/\/localhost:8183\/iiif\/2\/247%2Fimage-obj-datastream-359ebb5b-dcac-4537-8ad9-36a494572876.jp2",
                                "type": "ImageService2",
                                "profile": "level2"
                            }
                        ]
```

Reply:
I got it working with your help @DiegoPino ! 
I ended up with
```
            if (annotation.body.hasOwnProperty('service') && annotation.body.service[0]['id'] && isValidHttpUrl(annotation.body.service[0]['id'])) {
              imageObj.serviceUrl = annotation.body.service[0]['id'].replace(/\/$/, '');
            }
```

Reply:
JS is weird so just to be sure I would actually go for "has own property" for annotation ->  body ->service and then check if there is a single entry and then get if 'id' ... I know its a lot but... JS!

Reply:
Mostly because `replace` only works as a method for strings (or url)... so you might test it is a string?

Reply:
Added another sanity check. We don't need to test if the service url is a string, since isValidHttpUrl will return false if it isn't.